File: pyg-web/pyg-user-web/src/main/webapp/js/controller/userController.js
/** 定义控制器层 */
app.controller('userController', function($scope, baseService){

    // 定义用户对象
    $scope.user = {};

    // 定义保存用户方法
    $scope.save = function () {
        //  判断两次密码是否一致
        if ($scope.user.password != $scope.password){
            alert("两次密码不一致,请重新输入!");
            return;
        }
        baseService.sendPost("/user/save?smsCode=" + $scope.smsCode, $scope.user).then(function (response) {
            if (response.data){
                alert("注册成功!");
                $scope.user = {};
                $scope.password = "";
                $scope.smsCode = "";
            }else {
                alert("注册失败!");
            }
        });
    };

    $scope.sendCode = function () {
        if ($scope.user.phone && /^1[3|4|5|6|7|8|9]\d{9}$/.test($scope.user.phone)){
            baseService.sendGet("/user/sendCode?phone=" + $scope.user.phone).then(function (response) {
                alert(response.data ? "验证码已发送！" : "验证码发送失败！");
            });
        }else {
            alert("请输入手机号码!");
        }
    }
});

File: pyg-web/pyg-user-web/src/main/java/cn/pyg/user/controller/UserController.java
package cn.pyg.user.controller;

import cn.pyg.pojo.User;
import cn.pyg.service.UserService;
import com.alibaba.dubbo.config.annotation.Reference;
import org.springframework.web.bind.annotation.*;

/**
 * 用户控制器
 */
@RestController
@RequestMapping("/user")
public class UserController {

    @Reference(timeout = 1000000000)
    private UserService userService;

    @PostMapping("/save")
    public boolean save(@RequestBody User user, String smsCode){
        try{

           boolean ok =  userService.checkSmsCode(user.getPhone(), smsCode);
           if (ok){
               userService.save(user);
               return true;
           }
        }catch (Exception e){
            e.printStackTrace();
        }
        return false;
    }

    @GetMapping("/sendCode")
    public boolean sendCode(String phone){
        try{
            // 发送验证码
            userService.sendCode(phone);
            return true;
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }
}
